plugins/woocommerce-admin/client/customer-effort-score-tracks/customer-effort-score-exit-page.ts

```diff
@@ -0,0 +1,156 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { OPTIONS_STORE_NAME } from '@woocommerce/data';
+import { dispatch, resolveSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { ALLOW_TRACKING_OPTION_NAME } from './constants';
+
+const CUSTOMER_EFFORT_SCORE_EXIT_PAGE_KEY = 'customer-effort-score-exit-page';
+
+let allowTracking = false;
+resolveSelect( OPTIONS_STORE_NAME )
+	.getOption( ALLOW_TRACKING_OPTION_NAME )
+	.then( ( trackingOption ) => {
+		allowTracking = trackingOption === 'yes';
+	} );
+
+/**
+ * Gets the list of exited pages from Localstorage.
+ */
+export const getExitPageData = () => {
+	if ( ! window.localStorage ) {
+		return [];
+	}
+
+	const items = window.localStorage.getItem(
+		CUSTOMER_EFFORT_SCORE_EXIT_PAGE_KEY
+	);
+	const parsedJSONItems = items ? JSON.parse( items ) : [];
+	const arrayItems = Array.isArray( parsedJSONItems ) ? parsedJSONItems : [];
+
+	return arrayItems;
+};
+
+/**
+ * Adds the page to the exit page list in Localstorage.
+ *
+ * @param {string} pageId of page exited early.
+ */
+export const addExitPage = ( pageId: string ) => {
+	if ( ! window.localStorage ) {
+		return;
+	}
+
+	let items = getExitPageData();
+
+	items = items.filter( ( item ) => item !== pageId );
+	items.push( pageId );
+	items = items.slice( -10 ); // Upper limit.
+
+	window.localStorage.setItem(
+		CUSTOMER_EFFORT_SCORE_EXIT_PAGE_KEY,
+		JSON.stringify( items )
+	);
+};
+
+/**
+ * Removes the passed in page id from the list in Localstorage.
+ *
+ * @param {string} pageId of page to be removed.
+ */
+export const removeExitPage = ( pageId: string ) => {
+	if ( ! window.localStorage ) {
+		return;
+	}
+
+	let items = getExitPageData();
+
+	items = items.filter( ( item ) => item !== pageId );
```

Comment:
What do you think about renaming `item` to `pageExitedId` or something less generic?